### PR TITLE
Problem details cleanup

### LINKED ISSUE #34509: Problem Details related methods on Results should set the content type to "application/problem+json"
The methods `Microsoft.AspNetCore.Http.Results.Problem` and `Microsoft.AspNetCore.Http.Results.ValidationProblem` directly return a new `ObjectResult` response but do not change the content type to "application/problem+json" so the default content type of "application/json" is set instead.

### PR DESCRIPTION
**Pull request by pranavkm**
* Move HttpValidationProblemDetails to Microsoft.AspNetCore.Http per API review
* Return Content-Type headers when returning ProblemDetails responses

Fixes #34509

### CODE REVIEW

File: src/Http/Http.Results/src/ObjectResult.cs [hunk 1]
```diff
@@ -61,7 +66,7 @@ public Task ExecuteAsync(HttpContext httpContext)
             }
 
             OnFormatting(httpContext);
-            return httpContext.Response.WriteAsJsonAsync(Value);
+            return httpContext.Response.WriteAsJsonAsync(Value, Value.GetType(), options: null, contentType: ContentType);
```
Comment (davidfowl) [inline src/Http/Http.Results/src/ObjectResult.cs:69]: Ugh, callsites like this are what make trimming hard. I guess we'll need new result types and overloads to fix the problem.

File: src/Http/Http.Results/src/ObjectResult.cs [hunk 2]
```diff
@@ -37,6 +37,11 @@ public ObjectResult(object? value, int? statusCode)
         /// </summary>
         public int? StatusCode { get; set; }
 
+        /// <summary>
+        /// Gets the value for the <c>Content-Type</c> header.
+        /// </summary>
+        public string? ContentType { get; init; }
```
Comment (davidfowl) [inline src/Http/Http.Results/src/ObjectResult.cs:43]: Why is this `init;` but everything else is `set;`

**Review by davidfowl — APPROVED** [review]
Small nits

File: src/Http/Http.Results/src/ObjectResult.cs (hunk 2 above)
Comment (pranavkm) [reply src/Http/Http.Results/src/ObjectResult.cs:43]: No reason. I'll switch it to make it consistent
```suggestion
        public string? ContentType { get; set; }
```

File: src/Http/Http.Results/src/ObjectResult.cs (hunk 1 above)
Comment (pranavkm) [reply src/Http/Http.Results/src/ObjectResult.cs:69]: > I guess we'll need new result types and overloads

Sorry, I don't super follow.

Comment (davidfowl) [reply src/Http/Http.Results/src/ObjectResult.cs:69]: To allow for annotating things using the DAM attributes we either need a Type argument to flow or a generic argument to flow. That means ObjectResult needs a Type argument or it needs to be generic (ObjectResult\<T\>). Then you would mark those with the attributes to make sure stuff doesn't get trimmed but maybe this is premature.

**Pull request merged**